main/main.c

```diff
@@ -1858,6 +1858,7 @@ void php_request_shutdown(void *dummy)
 	 * inside zend_executor callback functions.
 	 */
 	EG(current_execute_data) = NULL;
+	EG(guard_context) = (uint32_t)-1;
```

Comment:
It may actually be better to do `EG(guard_context) = ++EG(guard_context_counter);` in `_zend_bailout()`, because that automatically works for the different sections wrapped in `zend_try` in `php_request_shutdown()`, each having an isolated context when the previous one was unsuccessful.